src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -265,155 +313,356 @@ export type RemoteConfig = {
+			return
+		end
+
+		if type(data.value) ~= "string" then
+			return
+		end
+
+		if type(data.dimension) ~= "string" then
+			return
+		end
+
+		self:SetCustomDimension({
+			userId = player.UserId,
+			dimension = data.dimension,
+			value = data.value,
 		})
 	end)
-	
+
 	MarketplaceService.PromptBundlePurchaseFinished:Connect(
 		function(player: Player, bundleId: number, purchased: boolean)
 			if not purchased then
 				return
 			end
-			
+
+			local productInfo = self:_getProductInfo(bundleId, Enum.InfoType.Bundle)
+
 			self:LogMarketplacePurchase({
 				userId = player.UserId,
 				itemType = "Bundle",
-				id = bundleId,
+				id = productInfo and productInfo.Name or bundleId,
 				cartType = "PromptPurchase",
+				robuxPrice = productInfo and productInfo.PriceInRobux,
 			})
 		end
 	)
-	
+
 	MarketplaceService.PromptGamePassPurchaseFinished:Connect(
 		function(player: Player, gamePassId: number, purchased: boolean)
 			if not purchased then
 				return
 			end
-			
+
+			local productInfo = self:_getProductInfo(gamePassId, Enum.InfoType.GamePass)
+
 			self:LogMarketplacePurchase({
 				userId = player.UserId,
 				itemType = "GamePass",
-				id = gamePassId,
+				id = productInfo and productInfo.Name or gamePassId,
 				cartType = "PromptPurchase",
+				robuxPrice = productInfo and productInfo.PriceInRobux,
 			})
 		end
 	)
-	
-	if self._options.logDevProductPurchases then
-		MarketplaceService.PromptProductPurchaseFinished:Connect(
-			function(player: Player, productId: number, purchased: boolean)
-				if not purchased then
-					return
-				end
-				
-				self:LogMarketplacePurchase({
-					userId = player.UserId,
-					itemType = "Product",
-					id = productId,
-					cartType = "PromptPurchase",
-				})
-			end
-		)
-	end
-	
-	MarketplaceService.PromptPurchaseFinished:Connect(
-		function(player: Player, assetId: number, purchased: boolean)
+
+	MarketplaceService.PromptProductPurchaseFinished:Connect(
+		function(userId: number, productId: number, purchased: boolean)
 			if not purchased then
 				return
 			end
-			
+
+			local productInfo = self:_getProductInfo(productId, Enum.InfoType.Product)
+
 			self:LogMarketplacePurchase({
-				userId = player.UserId,
-				itemType = "Asset",
-				id = assetId,
+				userId = userId,
+				itemType = "Product",
+				id = productInfo and productInfo.Name or productId,
 				cartType = "PromptPurchase",
+				robuxPrice = productInfo and productInfo.PriceInRobux,
 			})
 		end
 	)
-	
+
+	MarketplaceService.PromptPurchaseFinished:Connect(function(player: Player, assetId: number, purchased: boolean)
+		if not purchased then
+			return
+		end
+
+		local productInfo = self:_getProductInfo(assetId, Enum.InfoType.Asset)
+
+		self:LogMarketplacePurchase({
+			userId = player.UserId,
+			itemType = "Asset",
+			id = productInfo and productInfo.Name or assetId,
+			cartType = "PromptPurchase",
+			robuxPrice = productInfo and productInfo.PriceInRobux,
```

Comment:
Is LogMarketplacePurchase designed to accept robuxPrice as nil? If productInfo is nil for some reason, robuxPrice will be nil instead of 0.